source/agora/common/TransactionManager.d

```diff
@@ -0,0 +1,283 @@
+/*******************************************************************************
+
+    Contains class for transaction verification
+
+    Copyright:
+        Copyright (c) 2019 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.common.TransactionManager;
+import agora.common.Data;
+import agora.common.crypto.Key;
+import agora.common.Transaction;
+import std.digest.sha;
+import vibe.data.json : serializeToJsonString;
+
+/*******************************************************************************
+
+Class for the transactions verification
+
+*******************************************************************************/
+
+public class TransactionManager
+{
+    /// Store of previously processed transactions
+    private Transaction[Hash] txs;
+
+    /// null value of transaction hash
+    public static Hash nullTxHash;
```

Comment:
I see that this is never initialized. So it always have the default value, which can be represented as `Hash.init`.
It is better to use `Hash.init` instead of `nullTxHash`, as `Hash.init` can be optimized (https://en.wikipedia.org/wiki/Constant_folding) while `nullTxHash` cannot.